### PR TITLE
Polygon enemy/11 enemy ai

### PR DESCRIPTION
**Pull request by FearfulSnow**
# Changes
- Polygon enemy have a speed attribute that gets assigned when initilized. This attribute becomes smaller, the higher the edge provided. here the speed decreases logarithmically and ensures a much milder decrease as edges increases.

- Polygon enemy can now move towards the players coordinates using its speed attribute, regardless of a pathfindingProvider. 
but if a pathfindingProvider is provided the polygon enemy will move towards the coordinates provided by the pathfindingProvider.

# Related Issue
#11

### CODE REVIEW

File: PolygonEnemy/src/main/java/com/github/sef24sp4/polygonenemy/PolygonEnemyControlSystem.java
Comment (MidnightRocket) [inline PolygonEnemy/src/main/java/com/github/sef24sp4/polygonenemy/PolygonEnemyControlSystem.java:41]: Can optimize this code if we wait on #94

**Review by MidnightRocket — CHANGES_REQUESTED** [review]
There are still some unresolved issues from previous review. 
I also think it is good to wait until #94 is merged.

File: Player/src/main/java/module-info.java
Comment (FearfulSnow) [reply Player/src/main/java/module-info.java:1]: yeah just forgot you could do that ;)

**Comment by FearfulSnow** [comment]
@MidnightRocket can you look closer at the casting i am doing, because the functions return an IVector and the methods i want to use lies on basicVector, just want to make sure its not the wrong way.

**Comment by MidnightRocket** [comment]
> @MidnightRocket can you look closer at the casting i am doing, because the functions return an IVector and the methods i want to use lies on basicVector, just want to make sure its not the wrong way.

Luckily I had the foresight that this problem would arise. So I have made the static factory method [`BasicVector.from()`](https://github.com/SE-F24-SP4/semesterproject-4/blob/f3fb7e53d2d86b9db1a036ae994d26055d0cd4c2/Common/src/main/java/com/github/sef24sp4/common/vector/BasicVector.java#L28)

**Review by MidnightRocket — APPROVED** [review]

**Pull request merged**
